tests/soak/test_checksum.py

```diff
@@ -67,16 +65,17 @@ class SoakOpcodes(unittest.TestCase):
     @parameterized.expand(test_data_parameterized # , name_func=custom_name_func
     )
```

Comment:
```python
def setUp(self):
    os.makedirs(os.path.join(os.getenv("TRAVIS_BUILD_DIR"), "build"), exist_okay=True)
```

Reply:
I don't understand, why would it make sense to create $TRAVIS_BUILD_DIR/build, you mean setUp a single temporary directory?

Reply:
my bad.

```
def setUp(self):
    os.makedirs("tmp", exist_okay=True)

Reply:
```
os.makedirs("tmp", exist_ok=True)
```

did it. But adding method to the class, causes no test to run whatsoever. So I did it like this

```diff
@@ -62,20 +61,19 @@ def custom_name_func(testcase_func, index, params):
     )
 
 class SoakOpcodes(unittest.TestCase):
-    @parameterized.expand(test_data_parameterized # , name_func=custom_name_func
-    )
+    @parameterized.expand(test_data_parameterized , name_func=custom_name_func)
     def test(self, path, expected):
-        with tempfile.TemporaryDirectory() as tmpdir:
-            name = os.path.basename(path).replace(".csd.json", "")
-            csd = path.replace(".json", "")
-            rendered_file_path = os.path.join(tmpdir, name + ".wav")
-            subprocess.run(
-                [csound_path, csd, "-o", rendered_file_path] + cli_options,
-                check=True, env=env
-            )
-            with open(rendered_file_path, "rb") as handle:
-                binary_data = handle.read()
-            hasher = sha256()
-            hasher.update(binary_data)
-            actual_hashsum = hasher.hexdigest()
-            self.assertEqual(actual_hashsum, expected)
+        os.makedirs("tmp", exist_ok=True)
+        name = os.path.basename(path).replace(".csd.json", "")
+        csd = path.replace(".json", "")
+        rendered_file_path = os.path.join("tmp", name + ".wav")
+        subprocess.run(
+            [csound_path, csd, "-o", rendered_file_path] + cli_options,
+            check=True, env=env
+        )
+        with open(rendered_file_path, "rb") as handle:
+            binary_data = handle.read()
+        hasher = sha256()
+        hasher.update(binary_data)
+        actual_hashsum = hasher.hexdigest()
+        self.assertEqual(actual_hashsum, expected)
```